src/dialogsearch.cpp

```diff
@@ -144,7 +144,21 @@ void DialogSearch::on_button_search_clicked()
     }
     else if (ui->radio_switch->isChecked())
     {
-        //TODO implement me
+        int switchID = ui->combo_switch->currentData().toInt();
+
+        search_predicate = [switchID](const RPG::EventCommand& com)
+        {
+            switch(com.code)
+            {
+                case Cmd::ControlSwitches:
+                    return (com.parameters[0] != 2 && com.parameters[1] == switchID) ||
+                        (com.parameters[0] == 1 && com.parameters[2] == switchID);
+                case Cmd::ConditionalBranch:
+                    return com.parameters[0] == 0 && com.parameters[1] == switchID;
+                default:
+                    return false;
```

Comment:
No, but that should be done. How it should look like in the result table if we want to mix it with the other results? Maybe something like this:

Map | Event | Event Page | Sourceline | Action
------| ----|  -----|   ----- | -----
4 | 2 | 3 | 0? | Precondition S[0001] = True

Other preconditions include variables and items, so these should also show in the result table when they appear in some precondition.